sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -308,12 +326,34 @@ def convert_feature(fx):
         if not 1 <= np.size(fxs) <= 2:
             raise ValueError('Each entry in features must be either an int, '
                              'a string, or an iterable of size at most 2.')
-        if kind != 'average' and np.size(fxs) > 1:
-            raise ValueError(
-                f"It is not possible to display individual effects for more "
-                f"than one feature at a time. Got: features={features}.")
+        # store the information if 2-way PD was requested with ICE to later
+        # warn or raise an error depending of the original value of `kind`
+        ice_for_two_way_pd.append(kind_plot != 'average' and np.size(fxs) > 1)
+
         tmp_features.append(fxs)
 
+    if (isinstance(kind, str) and any(ice_for_two_way_pd) and
```

Comment:
I don't think we should warn only when kind is a str and raise an error otherwise. It seems more natural to me to always warn and never error.

Reply:
The rational behind the choice:

* passing a `str` is blocker for the moment so you want the warning only;
* if one pass a `list`, the user deliberately make the choice of a plot and it looks fine to raise an error. Indeed, requesting specifically `individual` and getting the `average` with a warning might not be enough "educative" to highlight that something is wrong.

I am not 100% settled but I wanted to give the argumentation before deciding to make a potential change.

Reply:
I think we need to decide on warning vs error by assuming that the `list` syntax is already supported.

To me, there are 2 options: always raise, or always warn. Whether a list is passed or not seems irrelevant; I understand the point about educative purpose but to me, this point is valid in all cases, i.e. whether a list was  passed or not

Reply:
I see your point. The code will always warns.